homeassistant/components/tractive/__init__.py

```diff
@@ -124,10 +128,16 @@ async def _listen(self):
                     if event["message"] == "activity_update":
                         self._send_activity_update(event)
                     else:
-                        if "hardware" in event:
+                        if "hardware" in event and self._last_hw_time != event[
+                            "hardware"
+                        ].get("time"):
+                            self._last_hw_time = event["hardware"]["time"]
                             self._send_hardware_update(event)
 
-                        if "position" in event:
+                        if "position" in event and self._last_pos_time != event[
+                            "position"
+                        ].get("time"):
```

Comment:
This logic has a negative side effect. For example, we get an exception in the listener and the entities become `unavailable`.
```
2021-09-03 16:45:30 DEBUG (MainThread) [homeassistant.components.tractive] Tractive is not available. Internet connection is down? Sleeping 10 seconds and retrying
```
After w while the listener will return online:
```
2021-09-03 16:45:41 DEBUG (MainThread) [homeassistant.components.tractive] Tractive is back online
```
but the entities stay `unavailable` because the event has the same time as previous one.
```
2021-09-03 16:45:41 DEBUG (MainThread) [homeassistant.components.tractive] last hw time: 1630654299, event time: 1630654299
2021-09-03 16:45:41 DEBUG (MainThread) [homeassistant.components.tractive] last pos time: 1630654299, event time: 1630654299
```
Such a state of unavailability may last several dozen minutes.

Reply:
Good point.
I think this should solve it https://github.com/home-assistant/core/pull/55170/commits/5a5e301fffa1e2625f8f4166555febd477552ff9

Reply:
Yes, this will solve the problem. I'll investigate if we really need `get()` here:
```python
event["position"].get("time")
event["hardware"].get("time")
```

Reply:
I can confirm, `get()` is not needed here, `time` is always present.